app/pages/user-reservations/UserReservationsPage.js

```diff
@@ -2,91 +2,310 @@ import PropTypes from 'prop-types';
+
+    return filterCache && filterCache.page ? Number(filterCache.page) : 1;
+  }
+
+  get count() {
+    if (this.tab === TABS.UPCOMING) {
+      return this.state.upcomingReservation.count;
+    }
+
+    if (this.tab === TABS.PAST) {
+      return this.state.pastReservation.count;
+    }
+
+    return 0;
+  }
+
+  get pages() {
+    return Math.round(this.count / PAGE_SIZE);
+  }
+
+  get tab() {
+    return this.state.tab;
+  }
+
+  get reservations() {
+    if (this.tab === TABS.UPCOMING) {
+      return this.state.upcomingReservation.data;
+    }
+
+    if (this.tab === TABS.PAST) {
+      return this.state.pastReservation.data;
+    }
+
+    return [];
+  }
+
+  setModel = (name, data, cb) => {
+    const currentState = get(this.state, name, {});
+
+    this.setState({
+      [name]: {
+        ...currentState,
+        ...data,
+      },
+    }, cb);
+  }
+
+  loadModel = (name, params, mapResponseToState, stateName) => {
+    const nameInState = stateName || name;
+
+    this.setModel(nameInState, {
+      loading: true,
     });
+
+    client.get(name, params)
+      .then((response) => {
+        const defaultNextState = {
+          loading: false,
+          data: get(response.data, 'results', []),
+        };
+        const customNextState = mapResponseToState ? mapResponseToState(response) : {};
+
+        this.setModel(nameInState, {
+          ...defaultNextState,
+          ...customNextState,
+        });
+      });
+  }
+
+  fetchReservations = (namespace, getFilters) => {
+    const filters = getFilters ? getFilters(this.filters) : this.filters;
+    const params = {
+      ...filters,
+      // Ignore default date search parameter because this view is not
+      // date specific.
+      date: undefined,
+      page_size: PAGE_SIZE,
+      include: 'resource_detail',
+      is_own: true,
+    };
+
+    this.loadModel('reservation', params, response => ({
+      count: response.data.count,
+    }), `${namespace}Reservation`);
+  };
+
+  loadUpcomingReservations = (getFilters) => {
+    const getFilterWithDefaults = (defaultFilters) => {
+      const filters = { ...defaultFilters, ordering: 'begin' };
+
+      return getFilters ? getFilters(filters) : filters;
+    };
+
+    this.fetchReservations('upcoming', getFilterWithDefaults);
+  }
+
+  loadPastReservations = (getFilters) => {
+    const getFilterWithDefaults = (defaultFilters) => {
+      const now = moment().format('YYYY-MM-DD[T]HH:mmZZ');
+      const filters = {
+        ...defaultFilters,
+        all: true,
+        ordering: '-begin',
+        // Note that using this method to fetch a reservation has a
+        // weakness: it'll also list currently ongoing reservations. By
+        // common heuristics these should not be considered to be in the
+        // past, but I could not find a more efficient way to find past
+        // reservations.
+        start: moment(ARBITRARY_START_DATETIME, 'YYYY-MM-DD[T]HH:mm').format('YYYY-MM-DD[T]HH:mmZZ'),
```

Comment:
`YYYY-MM-DD[T]HH:mmZZ` could be replaced with DATETIME_FORMAT constant.

Reply:
Good suggestion👍